### DISCUSSION TITLE
Automatic LDAP user creation by default as role manager

### DISCUSSION THREAD

**Discussion by MeinhartEsrohr** [kanboard/kanboard#5182] [Installation & Configuration]
Hello Community,

at the moment my users are created automatically after successful authentication via LDAP, they get the role "user", but I want them to get the role "manager" by default. I have tried many things now, but I can not get it to work.

Does anyone else have an idea ? 

my Config:
define('LDAP_AUTH', true);
define('LDAP_SERVER', 'ldap://192.168.1.1');
define('LDAP_PORT', 389);
define('LDAP_SSL_VERIFY', false);
define('LDAP_START_TLS', false);
define('LDAP_USERNAME_CASE_SENSITIVE', false);
define('LDAP_BIND_TYPE', 'proxy');
define('LDAP_USERNAME', 'ldapadmin');
define('LDAP_PASSWORD', 'PasswordforUserldapadmin');
define('LDAP_USER_BASE_DN', 'OU=users,DC=company,DC=Local');
define('LDAP_USER_FILTER', '(&(objectClass=user)(sAMAccountName=%s))');
define('LDAP_USER_ATTRIBUTE_USERNAME', 'sAMAccountName');
define('LDAP_USER_ATTRIBUTE_FULLNAME', 'cn');
define('LDAP_USER_ATTRIBUTE_EMAIL', 'mail');
define('LDAP_USER_ATTRIBUTE_GROUPS', 'memberof');
define('LDAP_USER_ATTRIBUTE_PHOTO', '');
define('LDAP_USER_ATTRIBUTE_LANGUAGE', '');
define('LDAP_USER_CREATION', true);
define(‘LDAP_USER_DEFAULT_ROLE_MANAGER’, true);
define('LDAP_GROUP_ADMIN_DN', '');
define('LDAP_GROUP_MANAGER_DN', 'OU=users,DC=company,DC=Local');
define('LDAP_GROUP_PROVIDER', false);
define('LDAP_GROUP_BASE_DN', '');
define('LDAP_GROUP_FILTER', '');
define('LDAP_GROUP_USER_FILTER', '');
define('LDAP_GROUP_ATTRIBUTE_NAME', 'cn');

Thank you for your answers

**Comment by fguillot** [comment]
You have to use LDAP group sync as documented here: https://docs.kanboard.org/v1/admin/ldap/#ldap-group-synchronization

**Comment by rafacamargo123** [comment]
@fguillot I believe this is a problem, because the setting "LDAP_USER_DEFAULT_ROLE_MANAGER" should work even without LDAP group sync, because this setting was introduced in #4384 as a way to every new user to be a manager.
It seems easy to fix, I've opened PR #5152 proposing a fix

**Comment by rafacamargo123** [comment]
As I said on the PR, this isn't as simple as I thought, so my PR didn't correctly fix this.
Still, I think this issue may be reopened because LDAP_USER_DEFAULT_ROLE_MANAGER isnt working, and its behaviour should not depend on LDAP group sync settings.
